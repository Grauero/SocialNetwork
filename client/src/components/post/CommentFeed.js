import React from 'react';
import propTypes from 'prop-types';

import CommentItem from './CommentItem';

const ComponentFeed = ({ comments, postId }) => (
  comments.map(comment => (
    <CommentItem key={comment._id} comment={comment} postId={postId} />
  ))
);

ComponentFeed.propTypes = {
  comments: propTypes.instanceOf(Array).isRequired,
  postId: propTypes.string.isRequired
};

export default ComponentFeed;
