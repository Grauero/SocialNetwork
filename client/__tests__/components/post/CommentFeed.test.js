import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import CommentFeed from '../../../src/components/post/CommentFeed';

const props = {
  comments: [
    { _id: '1', comment: 'comment1' },
    { _id: '2', comment: 'comment2' }
  ],
  postId: 'postId'
};
const component = shallow(<CommentFeed {...props} />);

it('matches snapshot for 2 comments', () => {
  expect(toJSON(component)).toMatchSnapshot();
});
