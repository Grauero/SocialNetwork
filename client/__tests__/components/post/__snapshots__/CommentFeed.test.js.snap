// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot for 2 comments 1`] = `
Array [
  <Connect(CommentItem)
    comment={
      Object {
        "_id": "1",
        "comment": "comment1",
      }
    }
    key="1"
    postId="postId"
  />,
  <Connect(CommentItem)
    comment={
      Object {
        "_id": "2",
        "comment": "comment2",
      }
    }
    key="2"
    postId="postId"
  />,
]
`;
